### ISSUE TITLE
Specifying a --dir doesn't work

### ISSUE THREAD

**Issue by Nincodedo** [SKCraft/Launcher#42]
http://pastebin.com/T59VTqnW

The --dir option gets set twice, once by my call and a second time by the default. Unless I'm doing it wrong?

**Comment by sk89q** [comment]
So the problem is that the bootstrap (1) inserts --dir and (2) also copies remaining args and passes them to the real launcher.

It's probably best if it doesn't include --dir if it was specified by the user.

**Comment by Zethalion** [comment]
Hi @sk89q I am trying to make the bootstrapper portable as well. Using 4.3-SNAPSHOT I get the same error when using --dir argument. The --portable argument doesn't make the working directory of the bootstrapper the same as where the bootstrapper is located.

**Comment by sk89q** [comment]
Right now, I don't think the bootstrapper takes any command line arguments and the only way it detects protable mode is if it finds a portable.txt in the same folder as the boostrapper.

--portable and --dir are currently handled downstream on the actual launcher, and it doesn't really actually use --portable because it just defers to --dir for the directory (and defaulting to the current directory if --dir is not specified, which the boostrapper always does).

Does portable.txt work for you?

**Comment by Zethalion** [comment]
Yes, placing a portable.txt next to the bootstrapper works. However, I'd prefer to have a config option in bootstrap.properties to do this. If I read Bootstrap.java correctly, this should already be possible at line 62 but should occur before line 59. Or line 59 needs to be changed somehow.
